src/components/chat/action-processor.js

```diff
@@ -81,28 +81,46 @@ export default class ActionProcessor {
 		}
 	}
 
+	async _executeOnDataset(action, dataset, ...args) {
+		if (!this._validateDatasetExists(dataset)) {
+			return [ActionProcessor.RESPONSES.INVALID_DATASET];
+		}
+
+		if (dataset == "Everything") {
+			var results = []
+
+			for (const datasetObj of this._getAllDatasets()) {
+				results.push(await action(datasetObj.id, ...args))
```

Comment:
nit: this will work just fine, but you can also use `Promise.all` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all